tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/cli/dockerCli/core/generateCodesInLocal.ts

```diff
@@ -0,0 +1,62 @@
+import { existsSync } from "fs";
+import * as path from "path";
+import { spawn } from "child_process";
+import { initializeDockerTaskEngineContext, runTaskEngine } from "./dockerTaskEngine";
+import { doNotExitDockerContainer } from "./doNotExitDockerContainer";
+import { DockerContext } from "./DockerContext";
+import { getChangedPackageDirectory } from "../../../utils/git";
+
+export const sdkToRepoMap = {
+    js: 'azure-sdk-for-js',
+    python: 'azure-sdk-for-python',
+    go: 'azure-sdk-for-go',
+    java: 'azure-sdk-for-java',
+    '.net': 'azure-sdk-for-net'
+}
+
+export async function cloneRepoIfNotExist(context: DockerContext, sdkRepos: string[]) {
+    for (const sdkRepo of sdkRepos) {
+        if (!existsSync(path.join(context.workDir, sdkRepo))) {
+            const child = spawn(`git`, [`clone`, `https://github.com/Azure/${sdkRepo}.git`], {
+                cwd: context.workDir,
+                stdio: ['ignore', 'pipe', 'pipe'],
+            });
+            child.stdout.on('data', data => context.logger.log('cmdout', data.toString()));
+            child.stderr.on('data', data => context.logger.log('cmderr', data.toString()));
+            await new Promise((resolve) => {
+                child.on('exit', (code, signal) => {
+                    resolve({ code, signal });
+                });
+            });
+        }
+        context.sdkRepo = path.join(context.workDir, sdkRepo);
+    }
+}
+
+export async function generateCodesInLocal(dockerContext: DockerContext) {
```

Comment:
Can the work of generateCodesInLocal  become a GenerateCodesInLocalJob with the execute() function?
Then  generate, build, mocktest etc, could be a Task in this job.

Reply:
I have wrapped it in a Job. However, the task of generte, build, mocktest belongs to docker task engine, and I don't hope we invoke them in Job directly.